src/controllers/requestController.js

```diff
@@ -53,31 +53,34 @@ export default class requestController {
   // creating a request
   static async createRequest(req, res) {
     try {
-      let _response = '';
-      const newIdRoom = req.body.idRoom,
-        allRequests = await findRequestByRoomId(newIdRoom);
+      const { idRoom, hotelName, dateStart, dateEnd } = req.body,
+        { id, firstname, lastname } = req.userData,
+        hotel = await findHotelByName(hotelName);
+
+      if (!hotel) res.status(403).json({ message: res.__('Hotel does not exist!') });
+      const room = await findRooms(idRoom);
 
-      if (allRequests) _response = res.status(403).json({ message: res.__('Room id is already occupied!') });
-      const {
-          idRoom, dateStart, dateEnd
-        } = req.body,
-        userId = req.userData.id;
+      if (!room) res.status(403).json({ message: res.__('Room id does not exist!') });
+      const roomStatus = room.status,
+        roomIds = await mapRoomIds(id);
 
-      const existingRoom = await findRooms(idRoom);
+      const oneRequest = await roomIds.find((elem) => elem === idRoom);
 
-      if (!existingRoom) _response = res.status(400).json({ message: res.__('Room id does not exist') });
+      if (oneRequest || roomStatus === 'OCCUPIED') return res.status(400).json({ message: res.__('Room Id is already occupied!') });
 
-      const newRequest = {
-        idUser: userId,
+      const savedRequest = await request.create({
+        idUser: id,
+        requesterName: `${firstname} ${lastname}`,
         idRoom,
+        roomType: room.roomType,
+        hotelId: hotel.id,
+        hotelName,
+        location: hotel.location,
         dateStart,
         dateEnd
-      };
-      const savedRequest = await request.create(newRequest);
+      });
 
-      _response = res.status(201).json({ message: res.__('Request created successfully!'), savedRequest });
-
-      return _response;
+      res.status(201).json({ message: res.__('Request created successfully!'), savedRequest });
     } catch (error) {
       return res.status(500).json({ error: res.__('Internal server error!') });
```

Comment:
And may validate the data from the user cause it's constantly throwing internal server error while the data you suggested in the documentation, are provided as a string as you specified